core/src/main/java/com/linecorp/armeria/server/healthcheck/DefaultCpuHealthChecker.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.healthcheck;
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+/**
+ * Forked from <a href="https://github.com/micrometer-metrics/micrometer/blob/8339d57bef8689beb8d7a18b429a166f6595f2af/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/system/ProcessorMetrics.java">ProcessorMetrics.java</a> in the micrometer core.
+ */
+public class DefaultCpuHealthChecker implements HealthChecker {
+
+    private static final List<String> OPERATING_SYSTEM_BEAN_CLASS_NAMES = Arrays.asList(
+            "com.ibm.lang.management.OperatingSystemMXBean", // J9
+            "com.sun.management.OperatingSystemMXBean" // HotSpot
+    );
+
+    private final OperatingSystemMXBean operatingSystemBean;
+
+    private final Class<?> operatingSystemBeanClass;
+
+    private final Method systemCpuUsage;
+
+    private final double targetCpuUsage;
+
+    private final Method processCpuUsage;
+
+    private final double targetProcessCpuLoad;
+
+    /**
+     * Instantiates a new Default cpu health checker.
+     *
+     * @param cpuUsage the cpu usage
+     * @param cpuIdle the cpu idle
+     * @param processCpuUsage the process cpu usage
+     * @param processCpuIdle the process cpu idle
+     */
+    public DefaultCpuHealthChecker(final int cpuUsage, final int cpuIdle,
+                                   final int processCpuUsage, final int processCpuIdle) {
+        this.targetCpuUsage = (double) cpuUsage / cpuIdle;
+        this.targetProcessCpuLoad = (double) processCpuUsage / processCpuIdle;
+        this.operatingSystemBean = ManagementFactory.getOperatingSystemMXBean();
+        this.operatingSystemBeanClass = requireNonNull(getFirstClassFound(OPERATING_SYSTEM_BEAN_CLASS_NAMES));
+        this.systemCpuUsage = requireNonNull(detectMethod("getSystemCpuLoad"));
+        this.processCpuUsage = requireNonNull(detectMethod("getProcessCpuLoad"));
+    }
+
+    @Override
+    public boolean isHealthy() {
+        final double currentSystemCpuUsage = invoke(systemCpuUsage);
+        final double currentProcessCpuUsage = invoke(processCpuUsage);
+        return currentSystemCpuUsage <= targetCpuUsage && currentProcessCpuUsage <= targetProcessCpuLoad;
+    }
+
+    private double invoke(final Method method) {
+        try {
+            return (double) method.invoke(operatingSystemBean);
```

Comment:
This method will be frequently invoked whenever `HealthCheckService` receives a request, hence [`MethodHandles`](https://www.baeldung.com/java-method-handles) might be a better choice.